mongo/collection.go

```diff
@@ -319,7 +319,8 @@ func (coll *Collection) insert(ctx context.Context, documents []interface{},
 //
 // The document parameter must be the document to be inserted. It cannot be nil. If the document does not have an _id
 // field when transformed into BSON, one will be added automatically to the marshalled document. The original document
-// will not be modified. The _id can be retrieved from the InsertedID field of the returned InsertOneResult.
+// will not be modified. The _id can be retrieved from the InsertedID field of the returned InsertOneResult. Note that
```

Comment:
The wording here should be stronger. We need to explicitly say that users should not do this. Same applies for the `InsertMany` function.

Reply:
Sounds good! Used some more force 💪